src/api.jl

```diff
@@ -140,18 +140,34 @@ function infer(ra_range::Tuple{Float64, Float64},
 
     # initialize tiled images and model parameters for trimming.  We will
     # initialize the psf again before fitting, so we don't do it here.
-    tiled_images, mp_all = ModelInit.initialize_celeste(images, catalog,
-                                                        tile_width=TILE_WIDTH,
-                                                        fit_psf=false)
+    info("initializing celeste without PSF fit")
+    tiled_images = SkyImages.break_blob_into_tiles(images, TILE_WIDTH)
+    mp = ModelInit.initialize_model_params(tiled_images, images, catalog,
+                                           fit_psf=false)
+
+    # get indicies of all sources relevant to those we're actually
+    # interested in, and fit a local PSF for those sources (since we skipped
+    # fitting the PSF for the whole catalog above)
+    info("fitting PSF for all relevant sources")
+    relevant_idx = ModelInit.get_all_relevant_sources(mp, idx)
+    for j in 1:size(mp.patches, 2)  # loop over images
+        for i in relevant_idx  # loop over relevant sources
+            patch = mp.patches[i, j]
+            psf = SkyImages.get_source_psf(patch.center, images[j])
+            mp.patches[i, j] = ModelInit.SkyPatch(patch, psf)
+        end
+    end
 
     results = Dict{Int, Dict}()
     for i in idx
```

Comment:
Just FYI, elsewhere in the code I have (tried to) consistently used "s" for indexing into sources and "sa" for indexing into active sources.  This improves readability a lot IMO.  You don't have to change it for this PR, but maybe consider it in the future.

Remind me to tell you sometime about a major bug at Google that went undetected for months because of non-descriptive index variable names.